### PR TITLE
[AIRFLOW-81] Add ScheduleBlackoutSensor

### PR DESCRIPTION
**Pull request by andscoop**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-81
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
After reviewing some of the older jira issues, I found one that could be simply solved via a sensorOperator as Chris suggested in the comments. This operator and test should be good enough to close out an ancient issue that hasn't gotten a lot of traction

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

Adds three test which validate the functionality of sensor properly returning true, false and error handling.

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality

- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by r39132** [comment]
@andscoop In your PRs, please replace `AIRFLOW-XXX` with the actual Jira issue this PR addresses, which is `AIRFLOW-81` in this case. This way, we can link a PR to a JIRA. Pls update your other Jiras accordingly.

**Comment by r39132** [comment]
@criccomini What do you think of this? There is definitely a need for this type of functionality (e.g. scheduled maintenance windows). Is a sensor the way to go? Won't we be blocking celery workers?

CC @bolkedebruin @Fokko

File: docs/code.rst [hunk 1]
```diff
@@ -222,6 +222,7 @@ Sensors
 .. autoclass:: airflow.contrib.sensors.pubsub_sensor.PubSubPullSensor
 .. autoclass:: airflow.contrib.sensors.qubole_sensor.QuboleSensor
 .. autoclass:: airflow.contrib.sensors.redis_key_sensor.RedisKeySensor
+.. autoclass:: airflow.contrib.sensors.sftp_sensor.ScheduleBlackoutSensor
```
Comment (boxysean) [inline docs/code.rst:225]: I believe this should be `airflow.contrib.sensors.ScheduleBlackoutSensor`.

**Review by boxysean — APPROVED** [review]
Thank you!

**Comment by bolkedebruin** [comment]
I think this is better solved by allowing the interval of a dag to have gaps. Pendulum (the underlying lib) allows for this afaik

**Comment by ron819** [comment]
I must say that it would be nice if there will be a way that the "blackout" won't be Airflow wide.. For example if it would be possible to exclude specific pools that will continue to be scheduled.

**Comment by Fokko** [comment]
The blackout should be on dag level. For example block the execution for the interval.

I fully agree with @bolkedebruin that we should use a smarter way of defining our periods. As Bolke is mentioning, Airflow is using Pendulum under the hood, and it would be nice to leverage the Period: https://pendulum.eustace.io/docs/#period

What do you think @andscoop?

**Comment by andscoop** [comment]
@Fokko I think that is the correct way to solve this issue - with the trade off that it will involve

**Comment by andscoop** [comment]
@Fokko I think that is the more complete solution in this case, it will just involve more thought around `start_date` and `end_date` and how they interact with the new "periods" feature described above. While potentially not significant, this new feature will likely add more logic to the scheduler loop and that trade-off needs to be considered.

With the maintainers' blessing, I can take this one and move forward with the feature as discussed.

**Comment by Fokko** [comment]
@andscoop Why would this make it more complex?

**Pull request closed without merging**